### ISSUE TITLE
TTS : aria-labelledby and aria-describedby not read

### ISSUE THREAD

**Issue by civodulab** [edrlab/thorium-reader#1933]
Hello,
Only the `aria-label`  is read ?
And in an SVG only `<title>` is read, the `<desc>` tag is not read.

**Comment by danielweck** [comment]
Hello, thank you for reporting this.
For SVG islands inside (X)HTML documents, Thorium uses the value of the `aria-label` attribute (if any) located on the root element of the SVG fragment, or fallback on the first `title` child element found beneath the first-descendants level (i.e. not recursively, just the top-level graphical shapes). Currently the `desc` tag is not supported. If I understand correctly, both `title` ( https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title ) and `desc` ( https://developer.mozilla.org/en-US/docs/Web/SVG/Element/desc ) attributes can appear multiple times in the SVG descendants, as they apply to specific graphics/shapes within the SVG sub-tree. Without an authored `aria-describedby` that concatenates specific referenced text (e.g. title, followed by description of a particular shape), I think Thorium can reasonably continue to pick the first available title / desc value (i.e. not to aggregate all available text in the SVG tree). So I suppose the order of precedence should be: `aria-label` (single line of text), `aria-describedby` (possibly multiple lines of text), or otherwise a fallback to `title` (first only) and/or `desc` (first only).
Does this sound good?

**Comment by danielweck** [comment]
Note that for images (i.e. the `img` HTML element) Thorium only selects the `alt` attribute. I suppose we could adapt a similar "cascade" including `aria-label` and `aria-describedby`?

**Comment by danielweck** [comment]
Note that for MathML, Thorium selects the `alttext` attribute, and fallback to the textual content within the math fragment. However also note that when MathJax is activated (either at document authoring time, or by Thorium itself at document rendering time), then Thorium finds the `aria-label` which can typically be generated dynamically / automatically by the MathJax library, or Thorium fallback to the original MathML `alttext` logic.

**Comment by civodulab** [comment]
If I understand correctly I have to put my description either in `<title>` or in the `aria-label` of the SVG.

**Comment by danielweck** [comment]
Technical details: https://github.com/readium/r2-navigator-js/issues/76

**Comment by danielweck** [comment]
Also see https://github.com/daisy/epub-accessibility-tests/issues/84

**Comment by danielweck** [comment]
See
 https://github.com/edrlab/thorium-reader/issues/3523
